src/marlin_stubs/host/M115.cpp

```diff
@@ -43,14 +43,18 @@ void GcodeSuite::M115() {
     SERIAL_ECHOPGM("FIRMWARE_NAME:Prusa-Firmware-Buddy ");
     SERIAL_ECHOPGM(project_version_full);
     SERIAL_ECHOPGM(" (Github) SOURCE_CODE_URL:https://github.com/prusa3d/Prusa-Firmware-Buddy");
-#if PRINTER_IS_PRUSA_MK4
-    SERIAL_ECHOLNPGM(" PROTOCOL_VERSION:" PROTOCOL_VERSION " MACHINE_TYPE:"
-                     "MK4"
-                     " EXTRUDER_COUNT:" STRINGIFY(1) " UUID:" MACHINE_UUID); // extruder hardcoded due to mmu, should be dynamic in the future
+    SERIAL_ECHOLNPGM(" PROTOCOL_VERSION:" PROTOCOL_VERSION " MACHINE_TYPE:" PRINTER_MODEL
+                     " EXTRUDER_COUNT:"
+#ifdef HAS_MMU2()
+        // Devices with MMU2/MMU3 only have one physical extruder
+        STRINGIFY(1)
+#elif HAS_TOOLCHANGER()
+        // Retrieve the amount of physical extruders for devices with Toolchanger
+        PrusaToolChangerUtils::get_num_enabled_tools()
```

Comment:
This code will not compile, because you're trying to concatenate C string literals with this.

Reply:
Also the comment above this is pointless.

Reply:
Ok, I removed the comment for the xl. It did compile, but it wouldn't have printed out the number in human-readable form, used stringify now.